src/app/modules/verification/verification.controller.ts

```diff
@@ -412,7 +477,24 @@ export const _handleOtpVerification: ControllerHandler<
   return (
     FormService.retrieveFormById(formId)
       // Step 2: Verify the otp sent over by the client
-      .andThen(() => VerificationService.verifyOtp(transactionId, fieldId, otp))
+      .andThen(() => {
+        let getFieldFromTransactionFx
+        let incrementFieldRetriesFx
+        if (fieldId === PAYMENT_CONTACT_FIELD_ID) {
+          getFieldFromTransactionFx = getPaymentContactFieldFromTransaction
+          incrementFieldRetriesFx = incrementPaymentFieldRetriesWrapper
+        } else {
+          getFieldFromTransactionFx = getFieldFromTransaction
+          incrementFieldRetriesFx = incrementFormFieldRetriesWrapper
+        }
```

Comment:
As a small extra nit to the discussion, whether they are in controller or in service, even for 2 instance of the same code, use a function:

```typescript
function getFieldFunctions(fieldId) {
  let getFieldFromTransactionFx
  let incrementFieldRetriesFx
  if (fieldId === PAYMENT_CONTACT_FIELD_ID) {
    getFieldFromTransactionFx = getPaymentContactFieldFromTransaction
    incrementFieldRetriesFx = incrementPaymentFieldRetriesWrapper
  } else {
    getFieldFromTransactionFx = getFieldFromTransaction
    incrementFieldRetriesFx = incrementFormFieldRetriesWrapper
  }
  
  return { getFieldFromTransactionFx, incrementFieldRetriesFx }
}

// [ ... ]

const { getFieldFromTransactionFx, incrementFieldRetriesFx } = getFieldFunctions(fieldId)

```

Reply:
Hmm regarding this, I got rid of passing around the functions completely 🤔 I simply used `isPayment` as a flag after comparing the `fieldId` to the `PAYMENT_CONTACT_FIELD_ID` value to improve code reuse. Is using separate functions and passing them around using the way you suggested above is better? Thoughts? haha

Reply:
It's not necessarily better no; passing a boolean `isPayment` for the decision to be taken deeper is ok.

Reading the current version of the PR, `isPayment` really is a "property" of `fieldId`, so it still feels a little funny. I wonder if the conditional check can be moved even deeper to the model 🤔

Reply:
hmm actually you're right, maybe the conditional check can just be moved into the model layer with the service layer not having any knowledge of it 🤔